### PR TITLE
Belindas_android-01-71-add_product_page

### LINKED ISSUE #71: USER STORY: Add Product Page
As an admin, I will only want other admins or people with the same permissions to view the add product page so that students won't have access to that page.

- On the horizon: Protect page “Add product page” so only users can view and use page.

### PR DESCRIPTION
**Pull request by Robel-003**
Resolves #71 

- Updating the AddProduct page allowing users to add a product.
- The "Add Product" in the page button doesn't have any functionality. 
- The page is temporarily accessible to all users. 
- Eventually need to protect the page and the button in the home page only allowing admin to access and use the button and page.

Screenshot 2023-10-20 at 5 42 44 PM

Screenshot 2023-10-20 at 5 40 43 PM

### CODE REVIEW

**Review by Annelisebx — DISMISSED** [review]

File: app/src/main/java/com/example/belindas_closet/screen/AddProduct.kt [hunk 1]
```diff
@@ -28,27 +32,50 @@ import androidx.compose.runtime.getValue
...
 import com.example.belindas_closet.model.Product
 import com.example.belindas_closet.model.Sizes
+import androidx.compose.ui.platform.LocalContext
 
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun AddProductPage(navController: NavHostController) {
+
     var productName by remember { mutableStateOf("") }
```
Comment (taylorpapke) [inline app/src/main/java/com/example/belindas_closet/screen/AddProduct.kt:48]: I would like to change this to productType and have a drop down box with all the clothing categories. This should match our schema @ https://github.com/SeattleColleges/belindas-closet-nestjs/blob/main/src/products/schemas/product.schema.ts.

**Review by taylorpapke — CHANGES_REQUESTED** [review]
Looks great but I just want to change that first field. I think productName as a string got mixed into our project earlier on at some point when it was originally intended to be productType as a String[].

File: app/src/main/java/com/example/belindas_closet/screen/AddProduct.kt (hunk 1 above)
Comment (Robel-003) [reply app/src/main/java/com/example/belindas_closet/screen/AddProduct.kt]: I change the productName to productType adding these values (shirts, pants, dresses, shoes) and also created a dropdown menu when adding a new product. Only following change that would be needed is when a user is updating/editing a specific product, I would just need to create another dropdown menu for that. I can create an issue and start on that

**Review by taylorpapke — DISMISSED** [review]
Thanks for making these changes!

**Comment by taylorpapke** [comment]
@Robel-003 Looks like some conflicts have popped up on this one due to @tinpham5614's recent PR merge https://github.com/SeattleColleges/belindas-closet-android/pull/85

**Comment by Robel-003** [comment]
> @Robel-003 Looks like some conflicts have popped up on this one due to @tinpham5614's recent PR merge #85

just resolved the merge conflicts. If all is well, I can merge this branch

**Comment by taylorpapke** [comment]
Oh no! A couple more just popped up with the other merges that just happened.

**Comment by Robel-003** [comment]
> Oh no! A couple more just popped up with the other merges that just happened.

no worries, just resolved

**Review by taylorpapke — APPROVED** [review]

**Comment by theGaryLarson** [comment]
@taylorpapke @Robel-003 
I am unable to build this to confirm it works. Am I missing something?

AddProduct.kt -> unable to resolve ProductType.SHIRTS

**Comment by Robel-003** [comment]
> @taylorpapke @Robel-003 I am unable to build this to confirm it works. Am I missing something?
> 
> AddProduct.kt -> unable to resolve ProductType.SHIRTS image

it's the enum value, that enum value doesn't exist, you just have to change it to another enum value. 
But this branch might produce some errors because of the difference in variables names in the AddProduct file, like with productName and productType. I'll need to update those variable names across other pages

**Comment by theGaryLarson** [comment]
I went to review this still unable to build due to the enum and variable names. Until it can build (enum and names are updated) I don't feel comfortable merging this into main. Our codebase needs to compile. @taylorpapke are you saying that it is updated in main? If so @Robel-003  I'd rebase this branch onto main so reviewers can build successfully.

**Comment by brinkbrink** [comment]
Thanks @theGaryLarson I created a new issue to resolve this naming discrepancy before we merge to main. I messaged @Robel-003 and @taylorpapke about it on Slack.

**Comment by taylorpapke** [comment]
@Robel-003 one last conflict to resolve here

**Review by taylorpapke — APPROVED** [review]
LGTM! I would ideally like to get the product name (productInfoField) removed but on another PR. Perhaps we can remove it when we add the remaining fields for the other product attributes to this screen.

**Review by tinpham5614 — APPROVED** [review]
LGTM!

**Pull request merged**